customer-terminate-service/domain/src/main/java/se/inera/intyg/cts/domain/service/EraseDataForCareProvider.java

```diff
@@ -0,0 +1,97 @@
+package se.inera.intyg.cts.domain.service;
+
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import se.inera.intyg.cts.domain.model.EraseService;
+import se.inera.intyg.cts.domain.model.Termination;
+import se.inera.intyg.cts.domain.model.TerminationStatus;
+import se.inera.intyg.cts.domain.repository.CertificateBatchRepository;
+import se.inera.intyg.cts.domain.repository.TerminationRepository;
+
+public class EraseDataForCareProvider {
+
+  private final static Logger LOG = LoggerFactory.getLogger(EraseDataForCareProvider.class);
+
+  private final List<EraseDataInService> services;
+  private final CertificateBatchRepository certificateBatchRepository;
+  private final TerminationRepository terminationRepository;
+
+  public EraseDataForCareProvider(List<EraseDataInService> services,
```

Comment:
Some javadoc for these methods would be nice. Kind of hard to follow with all private methods.

Reply:
Not sure that I agree... the private methods have meaningful names that say what they do. If we add javadoc to these methods it will be additional maintenance overhead as the javadoc always have to be consistent with the code and any internal refactoring of the class will break it.

Reply:
At least for me it would help to get a quick summary of the purpose each of these methods. Javadoc is best practise and used in all public libraries to help developers get a understanding of what the methods do. The overhead of updating is minimal and should be a part of the proccess. We can not asume that the purpose of the code is clear and future developers can quickly understand what the intended purpose is. One of my most common issues with working with older code is that the developers before me has made no dokumentation whatsoever and that creates a lot of extra work. It takes just a small amount of time to create a summary when you know and develop the code compared to trying to understand it 5 years later. :) The meaning of refactoring is to change the strukture of the code and not the behaviour so it should not affect the Javadoc and if it does it just takes a small amount of effort to update it. :)

Reply:
OK, I understand and get your point. I was mistaken before that you were asking for javadoc on the private methods.

I've written javadoc and tried to describe the behaviour of the method without going in to too much details.